service/controller/resource/spark/resource.go

```diff
@@ -81,3 +178,58 @@ func (r *Resource) toEncrypterObject(ctx context.Context, secretName string) (en
 
 	return enc, nil
 }
+
+func (r *Resource) getCredentialSecret(ctx context.Context, azureMachinePool key.LabelsGetter) (*v1alpha1.CredentialSecret, error) {
+	r.logger.LogCtx(ctx, "level", "debug", "message", "finding credential secret")
+
+	organization, exists := azureMachinePool.GetLabels()[label.Organization]
```

Comment:
You could use `label.Organization` from `apiextensions` here.

Reply:
I'll do a bulk change when we update apiextensions.